src/validation/transaction.cairo

```diff
@@ -29,6 +37,46 @@ pub fn validate_transaction(
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/consensus/tx_verify.cpp
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp
 
+    if (*tx.inputs).len() == 0 {
+        return Result::Err("transaction inputs are empty");
+    };
+    if (*tx.outputs).len() == 0 {
+        return Result::Err("transaction outputs are empty");
+    };
+
+    // https://github.com/bitcoin/bitcoin/blob/master/src/consensus/tx_verify.cpp#L17
+    let mut is_locktime_enabled = false;
```

Comment:
This is very not straightforward way to implement logic from `tx_verify.cpp`. I would prefer to keep Cairo code easily comparable to cpp. So I propose more or less:
```
fn is_final(tx: Transaction): Result<(), ByteArray> {
   if *tx.lock_time < LOCKTIME_THRESHOLD &&  *tx.lock_time < block_height {
            return Result::Ok(()))
   } 
   if *tx.lock_time >= LOCKTIME_THRESHOLD && *tx.lock_time < block_time {
            return Result::Ok(()))
   }

   for input in *tx.inputs {
          if *input.sequence != SEQUENCE_FINAL {     
              return Result::Err(
                  if tx.lock_time < LOCKTIME_THRESHOLD { 
                      "error msg with block_height " 
                  } else {
                      "error msg with block_time "                   
                  }
              )
          }
   };
}
```
It is important to keep directions of all operator the same. De Morgan laws is are a constant source or programming errors.